### ISSUE TITLE
md5sum verification goes wrong sometimes

### ISSUE THREAD

**Issue by rzbrk** [calliope-edu/CalliopEO_AstroPi#38]
In a couple of testcases md5sum is used to verify the correct output of the Calliope Mini after flashing with a dedicated hex.

Sometimes, the verification goes wrong with md5sum claiming, the files to verify are not in place:

```
-=# CalliopEO #=-
Unpacked /home/calliope/src/CalliopEO_AstroPi/30sec-counter.zip


programming: /home/calliope/src/CalliopEO_AstroPi/run_20210721-230028/30sec-counter.hex
done
open serial port
Calliope Mini found on /dev/ttyACM0  ..........
done
reading data

Start @ 2021/07/21-23:00:44; Will stop @ 2021/07/22-02:05:44

Send @START@

Received @START@
*****************
17 lines read
done
############################################################################################################

Check: Return code in all cases is 0 ... PASSED
Check: ZIP archive renamed to .done ... PASSED
Check: Folder run_* created ... PASSED
Check: Created two .data files ... PASSED
Check: Verify content of .data file ... md5sum: 30sec-counter.hex: No such file or directory
md5sum: 30sec-counter.hex.data: No such file or directory
md5sum: WARNING: 2 listed files could not be read
NOT PASSED
```
Reason currently unknown ...

The `sync` and `sleep 1` in all testcases after executing the Python script seems not to be sufficient.

**Comment by rzbrk** [comment]
Inserting a `ls -al` before executing md5sum in testcase 07 reveals that the heuristic approach in [this line](https://github.com/calliope-edu/CalliopEO_AstroPi/blob/da9fd3e8a05d08125a8a3d2b0576c1bb14aadf38/testcases/07_transm-timeout.sh#L131) to decide where to put the md5sum file and where to perform the md5sum verification can go wrong. This way, sometimes simply the wrong run_* folder is selected.

The following testcases are affected:
* 04_transmission-30s.sh
* 07_transm-timeout.sh
* 08_data-limit.sh

**Comment by rzbrk** [comment]
04_transmission-30s.sh not affected, because different code. There, we do not have to deal with different run_* folders.

**Comment by rzbrk** [comment]
Fixed in [edbda36](https://github.com/calliope-edu/CalliopEO_AstroPi/pull/39/commits/edbda362d8668279e87ff4cb79a06a2cf9589478).

Can be closed if #39 is merged.
